modules/lang-painless/src/main/java/org/elasticsearch/painless/antlr/PainlessParser.java

```diff
@@ -2834,8 +2837,21 @@ public final LambdaContext lambda() throws RecognitionException {
       }
       setState(448);
       match(ARROW);
-      setState(449);
-      block();
+      setState(451);
+      switch ( getInterpreter().adaptivePredict(_input,38,_ctx) ) {
```

Comment:
Nik is technically correct, but in this case were just trying to avoid the case where it has to fallback to picking the first rule because two full paths match.  As long as k != infinite (looking ahead all the tokens without the script) I think were fine.